### PR TITLE
Implement forgot password

### PR DESCRIPTION
**Pull request by gberaudo**
When a user forgets his password, he can:
- click on the forgot password in the authentication page;
- enter his email and submit request to the API;
- he receives a link by email and clicks on it;
- he enters a new password;
- the password is changed and the user logged in.

### CODE REVIEW

**Comment by gberaudo** [comment]
Depend on https://github.com/c2corg/v6_api/pull/204.

File: c2corg_ui/static/js/auth/auth.js [hunk 1]
```diff
@@ -173,8 +196,30 @@ app.AuthController.prototype.register = function() {
...
+  var alerts = this.alerts_;
+  /**
+   * @typedef {{
+   *   email: string
+   * }}
+   */
+  var data = this.scope_['requestChangePassword'];
+  this.api_.requestPasswordChange(data.email).then(function() {
+    var msg = alerts.gettext(
+        'We sent you an email, please click on the link to reset password.');
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/auth.js]: Perhaps you could add this string to https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/templates/i18n.html to make it add to the strings to translate.

**Comment by asaunier** [comment]
1 minor comment, else looks good to me, thanks!

File: c2corg_ui/static/js/auth/auth.js (hunk 1 above)
Comment (gberaudo) [inline c2corg_ui/static/js/auth/auth.js]: This should already be the case thanks to the call to `alerts.gettext`.

Comment (asaunier) [inline c2corg_ui/static/js/auth/auth.js]: OK, my mistake!

**Comment by asaunier** [comment]
@gberaudo feel free to merge the PR.

**Pull request merged**